ui/src/dashboards/selectors/index.ts

```diff
@@ -15,6 +15,47 @@ export const getTimeRange = (state: AppState): TimeRange => {
   return state.ranges[contextID] || DEFAULT_TIME_RANGE
 }
 
+export const getTimeRangeWithTimezone = (state: AppState): TimeRange => {
+  const timeRange = getTimeRange(state)
+  const timeZone = getTimeZone(state)
+  // create a copy of the timeRange so as not to mutate the original timeRange
+  const newTimeRange = Object.assign({}, timeRange)
+  if (timeRange.type === 'custom' && timeZone === 'UTC') {
+    // check to see if the timeRange has an offset
+    newTimeRange.lower = setTimeToUTC(newTimeRange.lower)
+    newTimeRange.upper = setTimeToUTC(newTimeRange.upper)
+  }
+  return newTimeRange
+}
+
+export const setTimeToUTC = (date: string): string => {
+  const offset = new Date(date).getTimezoneOffset()
+  // check if date has offset
+  if (offset === 0) {
+    return date
+  }
+  let offsetDate = date
+  if (offset > 0) {
+    // subtract tz minute difference
+    offsetDate = moment
+      .utc(date)
+      .subtract(offset, 'minutes')
+      .format()
+  }
+  if (offset < 0) {
+    // add tz minute difference
+    offsetDate = moment
+      .utc(date)
+      .add(offset, 'minutes')
+      .format()
+  }
+  return offsetDate
+}
```

Comment:
```suggestion
export const setTimeToUTC = (date: string): string => {
  const offset = new Date(date).getTimezoneOffset()
  if (offset > 0) {
    // subtract tz minute difference
    return moment
      .utc(date)
      .subtract(offset, 'minutes')
      .format()
  }
  if (offset < 0) {
    // add tz minute difference
   return moment
      .utc(date)
      .add(offset, 'minutes')
      .format()
  }
  // offset === 0
  return date
}
```

this is your code, but with a little bit further editing - the key was realizing that

```ts
if (offset === 0) {
  return date
}
let offsetDate = date

// if (offset > 0) ...
// if (offset < 0) ...

return offsetDate
```
could be turned into `return date` if it was moved below the `if` statements. the reason this change is worth making is that it removes one branch in an `if` statement, and removes the creation of a variable, and removes the overwriting of that variable and replaces them with a single return statement of the argument passed in. i've found that returning values directly, rather than returning them into a variable and returning makes the logic of function much easier to follow.

Reply:
Great suggestion! I'll make sure to push it up with the rest of the changes and ping you when it's up